Sources/searchable-collection-vc/SearchableCollectionViewLayout.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  SearchableCollectionViewLayout.swift
+//  Demo
+//
+//  Created by Yuliia Veresklia on 20/04/2017.
+//
+//
+
+import Foundation
+import UIKit
+
+class SearchableCollectionViewLayout: UICollectionViewFlowLayout {
+    
+    private var searchBackgroundColor = SearchBarView.defaultSearchBackgroundColor
+    
+    override init() {
+        super.init()
+        register(SearchBarView.self, forDecorationViewOfKind: SearchBarView.kind)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        var attributes = super.layoutAttributesForElements(in: rect)
+        
```

Comment:
I believe changing the `UICollectionViewLayoutAttributes` inside the array without copying them first is not allowed. You should do something like:

```
    let attributes = super.layoutAttributesForElementsInRect(rect)
    var attributesCopy = [UICollectionViewLayoutAttributes]()
    for itemAttributes in attributes! {
        let itemAttributesCopy = itemAttributes.copy() as! UICollectionViewLayoutAttributes
        // add the changes to the itemAttributesCopy                       
        attributesCopy.append(itemAttributesCopy)
    }

    return attributesCopy
```